File: Spark/PySpark DataFrame API/churn_model_agg_2.py
import sys
from pyspark.sql import *
import pyspark
from pyspark.sql import SparkSession
from pyspark.sql.types import *
from pyspark.sql.functions import *
import pyspark.sql.functions


if __name__ == "__main__":
    print("Churn Modelling")

# set the SparkSession
churn_df = SparkSession.builder.appName("Churn Modeling").master("local[3]").getOrCreate()

# read the datafile from the location
churn = churn_df.read.csv(r"D:\Code\DataSet\SparkDataSet\ChurnModeling.csv", header=True)
churn.show(5, truncate=False)

# select the required columns
churn_agg = churn.select("CustomerId", "Surname", "CreditScore", "Geography", "Gender", "Age", "Tenure") \
    .where(churn.Geography == "France") \
    .groupBy("Geography", "Gender") \
    .agg(avg("Age").alias("Avg_Age")) \
    .orderBy(asc("Avg_Age"))

churn_agg.show(15)
print(f'Records effected: {churn_agg.count()}')

File: Spark/Scala DataFrame API/churn_model_agg_2.scala
package Bank

//import Bank.HelloWorld.spark
import org.apache.spark.sql.SparkSession
import org.apache.spark.SparkContext
import org.apache.spark.SparkConf
import org.apache.spark.sql.SQLContext
//import org.apache.spark.sql.SQLContext.implicits._
//import sqlContext.implicits._

object ChurnModeling extends App {
  println("Hello Spark")
  val spark = SparkSession.builder
    .master("local[*]")
    .appName("Churn Modeling")
    .getOrCreate()

  //  read the datafile from the location
//  val sqlContext = new org.apache.spark.sql.SQLContext(sc)
  val churn = spark.read.option("inferSchema", "true").option("header", "true").csv("D:/Code/DataSet/SparkDataSet/ChurnModeling.csv")
  churn.show(10, truncate = false)

//  select the required columns
  val  churn_agg = churn.select("CustomerId", "Surname", "CreditScore",
    "Geography", "Gender", "Age", "Tenure")
    .filter("Geography == 'France'")
    .filter("CreditScore >= 501")
    .where("Gender == 'Male'" )
    .drop("Tenure")
    .show(10, truncate = false)
//  println(churn_agg.count())
}
